pkg/services/datasources/service/datasource.go

```diff
@@ -776,3 +778,14 @@ func (s *Service) CustomHeaders(ctx context.Context, ds *datasources.DataSource)
 	}
 	return s.getCustomHeaders(ds.JsonData, values), nil
 }
+
+func (s *Service) validateExternal() {
```

Comment:
This doesn't belong here I believe, becomes a bit strange relationship. Wouldn't you rather want to do this in plugins integration/plugins pkg after plugins have been loaded instead (whoever is calling the loader?

Reply:
yeah, probably not the best place. 

> whoever is calling the loader?

This would be the `pluginStore` but doesn't look like it's responsability either.

Another option (B) would be to create a new service that just does this. Seems a bit of an overkill for just logging an error but it would be self-contained.

Option C would be to refactor (again) the logic to load plugins so we go to the previous behavior in which all external plugins are loaded with a single plugin source but that code likes to break in all sort of unexpected ways so I am not eager to modify it again.

Thoughts @wbrowne ? 

I'm going to try with B for the moment.

Reply:
B looks promising. Thought it should stop the server first at failure, but you only want to log so this should be perfectly fine :+1:

Reply:
Btw, do you need/want to run the same validation after a plugin been installed during runtime?

Reply:
I think option B is definitely a better move for now 👍 

If we had a more strict definition of what core plugin paths were accepted to scan (right now external plugins are what necessitated the most recent change here), then we could create a specific loader pipeline for core plugins that included logic like this as a step in the same stage it was in before. Though I had imagined we may eventually need a "post initialzation" stage, which may also be more fitting for this case.

Reply:
> Thought it should stop the server first at failure, but you only want to log so this should be perfectly fine 👍

Yes, the idea of just one plugin missing / failing shouldn't stop the server (as we do for external plugins)

> Btw, do you need/want to run the same validation after a plugin been installed during runtime?

Since this affects to core plugins that cannot be installed through the catalog, we don't need to worry about installing them during runtime.

> If we had a more strict definition of what core plugin paths were accepted to scan (right now external plugins are what necessitated the most recent change here), then we could create a specific loader pipeline for core plugins that included logic like this as a step in the same stage it was in before. Though I had imagined we may eventually need a "post initialzation" stage, which may also be more fitting for this case.

The problem is that the check should be evaluated after all external plugins have been loaded so the pipeline for core plugins won't work here. The post-initialization step may be useful but only if it combines plugins from all pipelines executed (which seems cumbersome) :shrug: 

B. has been already implemented in this PR, please take another look

Reply:
>The problem is that the check should be evaluated after all external plugins have been loaded so the pipeline for core plugins won't work here. The post-initialization step may be useful but only if it combines plugins from all pipelines executed (which seems cumbersome) 🤷

Yeah I think we can rework in the future once we have a sane path approach - then all core plugins can be loaded together again - then we can have our pipeline step 😄 

>B. has been already implemented in this PR, please take another look

Great. Will do